src/System.Management.Automation/engine/CommandCompletion/PseudoParameterBinder.cs

```diff
@@ -1198,13 +1198,18 @@ private bool PrepareCommandElements(ExecutionContext context, CommandParameterAs
             string commandName = null;
             try
             {
+                context.IgnoreFailedScriptRequirements = true;
                 processor = PrepareFromAst(context, out commandName) ?? context.CreateCommand(commandName, dotSource);
```

Comment:
I see PrepareFromAst() uses a flag `forCompletion`. Maybe do the same for `context.CreateCommand()`?

Reply:
Yes, this can be merged. Only please review the last commit - many style issues.